app/src/main/java/de/westnordost/streetcomplete/data/osm/Countries.kt

```diff
@@ -0,0 +1,35 @@
+package de.westnordost.streetcomplete.data.osm
+
+import de.westnordost.countryboundaries.CountryBoundaries
+import de.westnordost.countryboundaries.getContainingIds
+import de.westnordost.countryboundaries.getIntersectingIds
+import de.westnordost.countryboundaries.isInAny
+import de.westnordost.osmapi.map.data.BoundingBox
+import de.westnordost.osmapi.map.data.LatLon
+import de.westnordost.streetcomplete.ktx.containsAny
+
+sealed class Countries
+
+object AllCountries : Countries()
+data class AllCountriesExcept(val exceptions: List<String>) : Countries() {
+    constructor(vararg exceptions: String) : this(exceptions.toList())
+}
+data class NoCountriesExcept(val exceptions: List<String>) : Countries() {
+    constructor(vararg exceptions: String) : this(exceptions.toList())
+}
+
+/** Whether these countries contain the given position.
+ *  Needs a CountryBoundaries to calculate that. */
+fun Countries.containsPosition(pos: LatLon, boundaries: CountryBoundaries) = when(this) {
+    is AllCountries -> true
+    is AllCountriesExcept -> !boundaries.isInAny(pos, exceptions)
+    is NoCountriesExcept -> boundaries.isInAny(pos, exceptions)
+}
```

Comment:
Hm, or 
```kotlin
/** whether the given position is in set of the given countries */
CountryBoundaries.isInAny(pos: LatLon, countries: Countries)
``` ?

"Needs a CountryBoundaries to calculate that" is maybe a sign that the function is a little counterintuitive.